src/systems/filecoin_mining/storage_mining/storage_miner_actor.go

```diff
@@ -293,7 +294,27 @@ func (sm *StorageMinerActor_I) verifySeal(onChainInfo sector.OnChainSealVerifyIn
 	// TODO: verify seal @nicola
 	// TODO: get var sealRandomness sector.SealRandomness from onChainInfo.Epoch
 	// TODO: sm.verifySeal(sectorID SectorID, comm sector.OnChainSealVerifyInfo, proof SealProof)
-	panic("TODO")
+
+	// verifySeal will also generate CommD on the fly from CommP and PieceSize
+
+	var pieceInfos []sector.PieceInfo // = make([]sector.PieceInfo, 0)
+
+	for dealId := range onChainInfo.DealIDs() {
+		// FIXME: Actually get the deal info from the storage market actor and use it to create a sector.PieceInfo.
+		_ = dealId
+
+		pieceInfos = append(pieceInfos, nil)
+	}
+
+	new(proving.StorageProvingSubsystem_I).VerifySeal(&sector.SealVerifyInfo_I{
```

Comment:
The actor wont have access to `proving.StorageProvingSubsystem_I` -- we should be calling something deeper, maybe `FilecoinProofsSubsystem.VerifySeal` (btw if that is a pure function, maybe it should be a static function (not bound to `FilecoinProofsSubsystem` type)

Reply:
since `filproofs` is a library, it doesn't need the `FilecoinProofsSubsystem` type. those functions can just be defined on the package directly.  (if `.id` doesn't allow, it can do it in the go file directly)

Reply:
I am expecting proofs methods to be exposed by the VMContext or similar, even if that just proxies directly to proofs implementations. I'm hoping actor code won't import *anything* outside of their own packages.

Reply:
Just a note that I simply needed to plumb this somehow and knew that VM details were not nailed down yet. Don't read anything into it, and I agree that there should be a context object serving as the single-point-of-contact between actors and the external world, including `filproofs`. If I missed the point, please clarify so I can best understand how to do these things in the future.